lightning/src/ln/interactivetxs.rs

```diff
@@ -118,34 +124,44 @@ impl AbortReason {
 
 impl Display for AbortReason {
 	fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
-		f.write_str(match self {
-			AbortReason::InvalidStateTransition => "State transition was invalid",
-			AbortReason::UnexpectedCounterpartyMessage => "Unexpected message",
-			AbortReason::ReceivedTooManyTxAddInputs => "Too many `tx_add_input`s received",
-			AbortReason::ReceivedTooManyTxAddOutputs => "Too many `tx_add_output`s received",
+		f.write_str(&match self {
+			AbortReason::InvalidStateTransition => "State transition was invalid".into(),
```

Comment:
Using `into()`  and `format!()` will allocate strings on the heap. We should avoid this by using `write_fmt()` instead of `write_str()` and use `format_args!()` around all the string literals.

Reply:
Reworked to avoid extra allocation; dynamic/static strings are handled in different matches now.